dhcpv4/ztpv4/parse_circuitid.go

```diff
@@ -0,0 +1,98 @@
+package ztpv4
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/insomniacslk/dhcp/dhcpv4"
+)
+
+// CircuitID represents the structure of network vendor interface formats
+type CircuitID struct {
+	Slot    string
+	Module  string
+	Port    string
+	SubPort string
+	Vlan    string
+}
+
+var circuitRegexs = []*regexp.Regexp{
+	// Juniper QFX et-0/0/0:0.0
+	regexp.MustCompile("^et-(?P<slot>[0-9]+)/(?P<mod>[0-9]+)/(?P<port>[0-9]+):(?P<subport>[0-9]+).*$"),
+	// Juniper PTX et-0/0/0.0
+	regexp.MustCompile("^et-(?P<slot>[0-9]+)/(?P<mod>[0-9]+)/(?P<port>[0-9]+).(?P<subport>[0-9]+)$"),
+	// Arista Ethernet3/17/1
+	regexp.MustCompile("^Ethernet(?P<slot>[0-9]+)/(?P<mod>[0-9]+)/(?P<port>[0-9]+)$"),
+	// Juniper QFX et-1/0/61
+	regexp.MustCompile("^et-(?P<slot>[0-9]+)/(?P<mod>[0-9]+)/(?P<port>[0-9]+)$"),
+	// Arista Ethernet14:Vlan2001
+	// Arista Ethernet10:2020
+	regexp.MustCompile("^Ethernet(?P<port>[0-9]+):(?P<vlan>.*)$"),
+	// Cisco Gi1/10:2020
+	regexp.MustCompile("^Gi(?P<slot>[0-9]+)/(?P<port>[0-9]+):(?P<vlan>.*)$"),
+	// Nexus Ethernet1/3
+	regexp.MustCompile("^Ethernet(?P<slot>[0-9]+)/(?P<port>[0-9]+)$"),
+	// Juniper bundle interface ae52.0
+	regexp.MustCompile("^ae(?P<port>[0-9]+).(?P<subport>[0-9])$"),
+}
+
+// ParseCircuitID will parse dhcpv4 packet and return CircuitID info
+func ParseCircuitID(packet *dhcpv4.DHCPv4) (*CircuitID, error) {
```

Comment:
We are going to use only the port number, aren't we? Does it make sense to return all the data or should we change this to:

```
func ParseCircuitID(packet *dhcpv4.DHCPv4) (int, error) {
```

Reply:
So yes,  we will be just using the port number. but I thought that as we are parsing the other info I should return all the fields so in future if anyone wants to use other fields they can use it.

Reply:
For creating a boot url we will need this all info. so I will be keeping it as CircuitID as a return type.

Reply:
Ok, thanks for the clarification.

Reply:
> For creating a boot url we will need this all info. so I will be keeping it as CircuitID as a return type.

Which data is needed for creating the boot url?

Reply:
Disregard, I just checked. You are right.